homeassistant/components/switchbot/coordinator.py

```diff
@@ -74,12 +77,12 @@ def _async_handle_bluetooth_event(
                 self._ready_event.set()
             _LOGGER.debug("%s: Switchbot data: %s", self.ble_device.address, self.data)
             self.device.update_from_advertisement(adv)
-        self.async_update_listeners()
+        super()._async_handle_bluetooth_event(service_info, change)
```

Comment:
We were actually writing state twice here.  We don't need to do that until the advertisement has been processed so I moved the super call to the end.

It wasn't noticeable until we added lights since the state could flip-flop